src/groups/circle_group.jl

```diff
@@ -167,9 +167,7 @@ function compose!(
 end
 
 identity_element(G::RealCircleGroup) = 0.0
-identity_element(::RealCircleGroup, p::AbstractArray) = map(i -> zero(eltype(p)), p)
-
-Base.inv(G::RealCircleGroup, p::AbstractArray{<:Any,0}) = map(pp -> inv(G, pp), p)
+identity_element(::RealCircleGroup, p) = zero(p)
```

Comment:
I don't remember why we had such an elaborate `identity_element(::RealCircleGroup, p::AbstractArray)` before but this looks fine.

But why did you delete that `inv`?

Reply:
The reason was that the first implementation was (following the general scheme in Matlab maybe?) both the Circle group a well as implicitly the PowerCirclegroup (implicitly working element-wise when you came along with vectors).

Inv for this group is generically defined (see that it even calls the generic `inv(G, pp)` but also line 172, the inv I deleted, is also from the interpretation of implicitly assuming it might also be its own power manifold.

The reason we changed that to explicitly require a power manifold is a) being more concise and b) exactly avoiding ambiguities like the oner we now had here left ;)

Reply:
Wasn't that `inv` method there to actually support points as zero-index arrays? Note that `inv` doesn't work on zero-index arrays. `inv(fill(1))` throws an error.

Reply:
Oh, maybe I missed the zero in the code, yes, you might be right.

Reply:
But we do not have a test case for that then – even more if we introduce this for all other functions (for example also `identity_element` we have a lot of ambiguities to fix.

Reply:
But we don't have any `inv` ambiguities currently as far as I can see and the `inv` method is covered according to Codecov.

Reply:
you're right, for inv we do not do the trait so we do not have that problem. I already reintroduced that line. We might just run into this if we have further functions where we have to consider special cases.